newrelic-agent/src/main/java/com/newrelic/agent/Agent.java

```diff
@@ -294,13 +282,24 @@ private static boolean isLicenseKeyEmpty(String licenseKey) {
 
     public static void main(String[] args) {
         String javaSpecVersion = JavaVersionUtils.getJavaSpecificationVersion();
-        if (!JavaVersionUtils.isAgentSupportedJavaSpecVersion(javaSpecVersion)) {
+        String sysExperimentalRuntime = System.getProperty("newrelic.config.experimental_runtime");
+        String envExperimentalRuntime = System.getenv("NEW_RELIC_AGENT_EXPERIMENTAL_RUNTIME");
```

Comment:
I believe that `System.getProperty` and `System.getenv` can potentially cause Exceptions that need to be guarded against.

https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getProperty-java.lang.String-
https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getenv-java.lang.String-

Also we shouldn't be hardcoding the the same strings in multiple places throughout the codebase (e.g. `"newrelic.config.experimental_runtime"` and `"NEW_RELIC_AGENT_EXPERIMENTAL_RUNTIME"`). It's better if they can be centralized into constants with the proper access level.

Reply:
Both  System.getProperty() and System.getenv() are effectively guarded by Boolean.parseBoolean()..because if parseBoolean(null) takes a null parameter then it will return false.  

` 
   public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
    
`

Reply:
After discussing with @twcrone on a review we decided to keep the string values for now. This pattern was chosen to go with how `agent_enabled` is done for the same classes.  The options needs to be set at a similar time and way as agent_enabled.

https://github.com/newrelic/newrelic-java-agent/blob/2cc6d9643e9060732940351950176e07172e824c/newrelic-agent/src/main/java/com/newrelic/bootstrap/BootstrapAgent.java#L98

If it is preferable to use a global static constant we decided that we should do that for each case in a separate PR to "clean-up" the usage in the same way for consistency.